samcli/lib/samlib/resource_metadata_normalizer.py

```diff
@@ -67,6 +70,19 @@ def normalize(template_dict):
                     },
                 )
 
+        if normalize_parameters and is_cdk_project(template_dict):
+            resources_as_string = json.dumps(resources)
+            parameters = template_dict.get("Parameters", {})
+
+            for parameter_name, parameter_value in parameters.items():
+                if (
+                    parameter_name.startswith("AssetParameters")
+                    and "Default" not in parameter_value
+                    and parameter_value.get("Type", "") == "String"
+                    and f'"Ref": "{parameter_name}"' not in resources_as_string
+                ):
+                    parameter_value["Default"] = " "
```

Comment:
Why we needed to set default to empty string to these parameters? Adding a log line could be helpful when debugging

Reply:
these are the auto generated CDK Asset parameters, I preferred to set default value instead of removing them to avoid removing a customer parameter that follow the same name pattern, so the customer can use sam deploy or package commands without providing values for these parameters although they are not used.

I will add some comments, and logging to this code

Reply:
added comments and debugging logs